scene/2d/audio_stream_player_2d.cpp

```diff
@@ -296,6 +297,10 @@ float AudioStreamPlayer2D::get_volume_db() const {
 }
 
 void AudioStreamPlayer2D::set_pitch_scale(float p_pitch_scale) {
+	if (p_pitch_scale < MIN_PITCH_SCALE) {
+		ERR_PRINT("The Pitch Value is bellow the minimum value and as such the audio will be paused");
```

Comment:
Why not add a return after the ERR_PRINT so that pitch_scale can't be set below the minimum? That way you can remove the check you added on _mix_audio.

Reply:
I though about that and it would definitely be more readable, the reason I didn't do that is so that peoples game doesn't crash if they are changing the pitch the same way i did to test. I thought it would be better. But that is my opinion.
If not crashing is not a need yeah that fix is much better as far as I see it.

Reply:
I added the check so that users could still set those low values for the pitch, and my test script would get stuck on low pitches due to the way I increment the pitch value, however I have tested the way you recommend and it solves the issue as well and if you believe that is the best way, Ill do it, I do believe the way you propose makes the code more readable.

Reply:
I think it's cleaner the way I say, and you can avoid the check at _mix_audio. But maybe somebody else could give another opinion on this too.

Reply:
considering nobody seems to be against your idea I am going to go ahead and implement it